### PR TITLE
Set-up project's basic

### PR DESCRIPTION
**Pull request by olegbespalov**
# What?

This pull-request should set up the project's basics, like:
* package.json
* build
* CI

# Why?

We'd like to establish a clear separation between the libraries' source-codes and hosted versions (minified).

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
+  "arrowParens": "always",
   "singleQuote": true,
-  "trailingComma": "all"
-}
+  "tabWidth": 2,
+  "printWidth": 80,
+  "trailingComma": "none"
```
Comment (mstoykov) [inline .prettierrc:6]: nit: why change this?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// this is the entry point for the library
+// it exports all the functions that are available to the user
+// it uses also for the bundling and minifying
+// (see package.json's "build" script)
+import { jUnit } from './junit.js';
+import { humanizeValue, textSummary } from './text.js';
+
+exports.jUnit = jUnit;
+exports.humanizeValue = humanizeValue;
+exports.textSummary = textSummary;
```
Comment (mstoykov) [inline src/index.js:10]:
```suggestion
export { jUnit } from './junit.js';
export { humanizeValue, textSummary } from './text.js';
```

1. this is ... more ESM-y
2. mixing commonjs and ESM is not good idea and soon will not be supported in k6 ... or at least not in a good way.

File: .prettierrc (hunk 1 above)
Comment (olegbespalov) [reply .prettierrc:6]: I copied this from one of another projects, but actually can easily revert this change or use the right config that is most commonly used in our projects :shrug:

Comment (olegbespalov) [reply .prettierrc]: Updated, used the https://github.com/grafana/k6-jslib-aws/blob/main/.prettierrc.json

Comment (mstoykov) [reply .prettierrc]: This does make quite a lot of whitespace changes and remove `;` at end and other strange things.

I wonder if we can shelf *that* particular change for now, just to merge this as is and then discuss this separately.

We do use https://github.com/grafana/k6-docs/blob/main/docs/sources/.prettierrc in the docs ... which I would expect is the one more people have seen and looked at.

File: .eslintrc [hunk 3]
```diff
@@ -6,8 +6,8 @@
       "ecmaVersion": 2020,
       "sourceType": "module"
     },
-    "extends": ["airbnb", "prettier"],
-    "plugins": ["react", "prettier"],
+    "extends": ["prettier"],
```
Comment (oleiade) [inline .eslintrc:9]: I recently discussed prettier and eslint configs with @EdvinasDaugirdas, and he recommended using the `eslint:recommended` base config and mentioned that's what they tend to default to in their team. We've been applying this to jslib-aws recently: https://github.com/grafana/k6-jslib-aws/pull/81/files#diff-e7e195d3d8fa5b82fa51ed952f3a179d429cbc46c0432ce1f7b357c54c162822R3

Maybe we want to do the same here?

```suggestion
    "extends": ["eslint:recommended"],
```

**Review by oleiade — APPROVED** [review]

File: .eslintrc (hunk 3 above)
Comment (olegbespalov) [reply .eslintrc]: @oleiade thanks! I've used that config, and the last commit contains the fixes required by this rule. Please check and let me know if they are fine :+1:

**Review by oleiade — APPROVED** [review]

**Review by mstoykov — APPROVED** [review]
LGTM!

Just don't forget to update the in k6 copy of the `text.js`

File: .eslintignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 .cache/
 public/
-node_modules/
\ No newline at end of file
+node_modules/
+build/
```
Comment (mp-dthr) [inline .eslintignore:4]: new line
```suggestion
build/

```

File: .eslintrc [hunk 5]
```diff
@@ -20,72 +20,7 @@
...
-            "data"
-          ]
-        }
-      ],
-      "import/prefer-default-export": 0,
-      "jsx-a11y/html-has-lang": 0, // We use react-helmet for that
-      "jsx-a11y/control-has-associated-label": 0
+      "no-underscore-dangle": "off"      
     }
   }
```
Comment (mp-dthr) [inline .eslintrc:26]:
```suggestion
  
  new line
```

File: .github/workflows/checks.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  push:
+    branches: [main]    
+  pull_request:
+    branches: [main]
+
+jobs:
+  checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```
Comment (mp-dthr) [inline .github/workflows/checks.yml:13]:
```suggestion
      - name: Checkout
        uses: actions/checkout@v4
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  push:
+    tags:
+      - v*
+
+jobs:
+  checks:
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/v')
+    steps:
+      - uses: actions/checkout@v4
```
Comment (mp-dthr) [inline .github/workflows/release.yml:13]:
```suggestion
      - name: Checkout
        uses: actions/checkout@v4
```

File: package.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    "devDependencies": {
+        "esbuild": "^0.14.51",
+        "eslint": "^8.20.0",
+        "eslint-config-prettier": "^8.5.0",
+        "eslint-plugin-prettier": "^4.2.1",
+        "prettier": "^2.7.1"
+    },
+    "engines": {},
+    "scripts": {
+        "lint": "eslint src",
```
Comment (mp-dthr) [inline package.json:15]: I would add max warnings to 0

**Review by mp-dthr — CHANGES_REQUESTED** [review]

**Comment by mp-dthr** [comment]
Also, one suggestion, you can add checks for the PR, i.e. block merging the PR until all comments are resolved, all jobs from the workflow passed, etc.

**Comment by olegbespalov** [comment]
@mp-dthr thanks! I've applied some of your suggestions, and we could iterate in the following PRs for another one. For now merging this as it is :+1:

**Pull request merged**